### ISSUE TITLE
Basic Test failing with `Your callback function returned a promise that never resolved.`

### ISSUE THREAD

**Issue by claytron5000** [abhinaba-ghosh/cypress-react-selector#42]
Test:
```
describe("Search input", () => {
	beforeEach(() => {
		cy.visit("/");
		cy.waitForReact();
	});

	context("viewport width === 640", () => {
		it("is visible", () => {
			cy.viewport(650, 1_024);
			cy.react("DashBoardHeader").should("be.visible");
		});
	});
});
```

cypress.json
```
{
    "pluginsFile": false,
    "supportFile": false,
    "baseUrl": "http://localhost:8080",
    "env": {
        "cypress-react-selector": {
            "root": "#root"
        }
    }
}
```
error:

```

assert
expected { Object () } to be visible
4
then
function(){}
CypressError
cy.then() timed out after waiting 4000ms.

Your callback function returned a promise that never resolved.

The callback function was:

(window) => {
    const isPrimitive = (x) =>
      Cypress..isNumber(x) || Cypress..isString(x) || Cypress._.isBoolean(x);

    const getNodes = () => {
      let elements;
      if (!window.resq) {
        throw new Error(
          '[cypress-react-selector] not loaded yet. did you forget to run cy.waitForReact()?'
        );
      }

      if (subject) {
        elements = window.resq.resq$(component, subject[0]);
      } else {
        if (getReactRoot(reactOpts.root) !== undefined) {
          elements = window.resq.resq$(
            component,
            document.querySelector(getReactRoot(reactOpts.root))
          );
        } else {
          elements = window.resq.resq$(component);
        }
      }

      if (reactOpts.props) {
        elements = elements.byProps(reactOpts.props);
      }
      if (reactOpts.state) {
        elements = elements.byState(reactOpts.state);
      }
      if (!elements.length) {
        return null;
      }

      let nodes = [];
      elements.forEach((elm) => {
        var node = elm.node,
          isFragment = elm.isFragment;
        if (isFragment) {
          nodes = nodes.concat(node);
        } else {
          nodes.push(node);
        }
      });

      return nodes;
    };

    const resolveValue = () => {
      return new Cypress.Promise.try(getNodes).then((value) => {
        if (!isPrimitive(value)) {
          value = Cypress.$(value);
        }
        return cy.verifyUpcomingAssertions(value, options, {
          onRetry: resolveValue,
        });
      });
    };

    return resolveValue()
      .then((value) => {
        return value;
      })
      .catch((err) => {
        throw new Error(
          getComponentNotFoundMessage(
            component,
            reactOpts.props,
            reactOpts.state
          )
        );
      });
  }Learn more
```

**Comment by abhinaba-ghosh** [comment]
hi @claytron5000 , I am aware of this issue. But fir me its an intermittent issue. I will raise a concern with Gleb tomorrow. 

can you let me know whats the behaviour if you run the same test multiple times? is it still throwing the same error?

**Comment by abhinaba-ghosh** [comment]
moreover, if you can provide a sample repository, i can take a deeper look.

**Comment by claytron5000** [comment]
@abhinaba-ghosh I want to keep this issue fresh, unfortunately I haven't had a chance to revisit this. Once I get time to revisit I'll create  an example/recreate the issue.

**Comment by abhinaba-ghosh** [comment]
Hi @claytron5000 , the issue should be resolved in `v2.1.0`. Feel free to re-open the ticket if you find the same in the future.

**Comment by axelprog** [comment]
This issue reproduced in `v2.3.6`  with material-ui 

I use the next code for test
```
  cy.react('NavBar')
      .react('NavItem', { props: { href: '/test'} })
      .should('be.visible');
```      

I tried use `be.visible`, `not.exist`, `have.length` in `.should()`. Also I tried change `NavBar` and `NavItem` with name property and in this case  works 
```
  cy.react('*', { props: { name: 'navigationBar' } })
      .react('*', { props: { name: 'navigationItem', href: '/test'}})
      .should('be.visible');
```
but it fails with the same error if I use `.should('not.exist');`  And the same behavior if I add ` exact:true` to the second `.react` call and selector can not find exactly `navigationItem` I looking for

![image](https://user-images.githubusercontent.com/3667905/111666274-9aaa8a00-8824-11eb-9871-7d875fbd5ace.png)
![image](https://user-images.githubusercontent.com/3667905/111666441-c75ea180-8824-11eb-8e17-5707f0bf41aa.png)
